### ISSUE TITLE
[BUG] 3.15.3 breaks union types returned from step.run

### ISSUE THREAD

**Issue by gwax** [inngest/inngest-js#513]
**Describe the bug**
Since 3.15.3, `step.run` with union return types are being given return type of `Promise<any>`

**To Reproduce**

1. Create file with:
```typescript
import { EventSchemas, Inngest } from "inngest";

type Thing = {
  data: {
    stuff?: string | null;
  };
};

type Events = {
  "thing/thing": Thing;
};

const inngest = new Inngest({
  id: "inngest",
  schemas: new EventSchemas().fromRecord<Events>(),
});

function logstuff(stuff: string | null) {
  console.log(stuff);
}

export default inngest.createFunction(
  {
    id: "thing-thing",
  },
  { event: "thing/thing" },
  async ({ step, event }) => {
    let { stuff } = event.data;
    stuff = await step.run("fix-undefined-stuff", () => {  // <- this is typed as (...) => Promise<any> should be (...) => Promise<string | null>
      let newStuff = stuff;
      if (newStuff === undefined) {
        newStuff = null;
      }
      return newStuff;  // <- type = string | null
    });

    await step.run("log-stuff", () => {
      logstuff(stuff);  // <-- This line is fine on 3.15.2 but type errors on 3.15.3
    });
  },
);
```
2. Typecheck with inngest==3.15.2 -> no errors
2. Typeschek with inngest==3.15.3 -> type is not narrowed by running "fix-undefined-stuff" step

**Expected behavior**
No type errors

**Code snippets / Logs / Screenshots**
```
[{
	"resource": ".../thing.ts",
	"owner": "typescript",
	"code": "2345",
	"severity": 8,
	"message": "Argument of type 'string | null | undefined' is not assignable to parameter of type 'string | null'.\n  Type 'undefined' is not assignable to type 'string | null'.",
	"source": "ts",
	"startLineNumber": 38,
	"startColumn": 16,
	"endLineNumber": 38,
	"endColumn": 21
}]
```

**System info (please complete the following information):**

- OS: Mac
- npm package Version 3.15.4

**Additional context**

**Comment by gwax** [comment]
Updated: the issue started in 3.15.3 NOT 3.15.4

**Comment by jpwilliams** [comment]
Thanks for the report, @gwax!

I've added a fix for this at #512. You can test it out now with:
```
npm install inngest@pr-512
```
Looks like our supported TS versions have grown wider than [unional/type-plus](https://github.com/unional/type-plus) that we've been using for the occasional helper type. The same happened recently with [sindresorhus/type-fest](https://github.com/sindresorhus/type-fest) in #500.

Let me know if the above works for you and I'll ship this tomorrow to fix.

**Comment by jpwilliams** [comment]
The above fix for this is shipped in [v3.15.5](https://github.com/inngest/inngest-js/releases/tag/inngest%403.15.5). 🙂

**Comment by gwax** [comment]
I can confirm this is fixed. Thank you.
